sklearn/feature_selection/sequential_feature_selector.py

```diff
@@ -0,0 +1,332 @@
+"""
+Sequential feature selection
+
+"""
+
+# Author: Sebastian Raschka <se.raschka@gmail.com>
+#
+# License: BSD 3 clause
+
+import numpy as np
+from itertools import combinations
+from .base import SelectorMixin
+from ..base import BaseEstimator
+from ..base import MetaEstimatorMixin
+from ..base import clone
+from ..utils.validation import check_is_fitted
+from ..model_selection import cross_val_score
+from ..metrics.scorer import check_scoring
+
+
+class SequentialFeatureSelector(BaseEstimator, MetaEstimatorMixin,
+                                SelectorMixin):
+    """Feature selector that selects features via greedy search.
+
+    This Sequential Feature Selector adds (forward selection) or
+    removes (backward selection) the features (X) to form a feature subset
+    in a greedy fashion that optimizes the extrinsic performance metric
+    of a Regressor or Classifier on the desired ouputs (y).
+
+    Read more in the :ref:`User Guide <sequential_feature_selection>`.
+
+    Parameters
+    ----------
+    estimator : scikit-learn Classifier or Regressor
+        Invoking the ``fit`` method on the `SequentialFeatureSelector`
+        will fit a clone of the original estimator that
+        will be stored in the class attribute `self.estimators_`.
+
+    n_features_to_select : int or tuple (default=1)
+        An integer arguments specifies the number of features to select,
+        where n_features_to_select < the full feature set.
+        Optionally, a tuple containing a min and max value can be provided
+        so that the feature selector will return a feature subset between
+        with min <= n_features <= max that scored highest in the evaluation.
+        For example, the tuple (1, 4) will return any combination from
+        1 up to 4 features instead of a fixed number
+        of features `n_features_to_select`.
+
+    scoring : str, callable, or None (default=None)
+        A string (see model evaluation documentation) or a scorer
+        callable object / function with signature `scorer(estimator, X, y)`.
+
+    forward : bool (default=True)
+        Performs forward selection if True and backward selection, otherwise.
+
+    cv : int or cross-validation generator, or an iterable (default=5)
+        Determines the cross-validation splitting strategy for
+        feature selection. Possible inputs for cv are:
+        - 0 or None, don't use cross validation
```

Comment:
why do we need both?

Reply:
What are you exactly referring to? I copied this doc string from`cross_val_score`

Reply:
Both 0 and None. Okay.